Gateway/Validator/CheckoutResponseValidator.php

```diff
@@ -67,28 +68,36 @@ public function validate(array $validationSubject)
         $isValid = true;
         $errorMessages = [];
         $resultCode = array_key_exists("resultCode", $response) ? $response['resultCode']:'';
+
         // validate result
-        if (!empty($resultCode)) {
+        if (!empty($response['resultCode'])) {
+            $resultCode = $response['resultCode'];
             $payment->setAdditionalInformation('resultCode', $resultCode);
+
+            if (!empty($response['action'])) {
+                $payment->setAdditionalInformation('action', $response['action']);
+            }
+
+            if (!empty($response['additionalData'])) {
+                $payment->setAdditionalInformation('additionalData', $response['additionalData']);
+            }
+
+            if (!empty($response['pspReference'])) {
+                $payment->setAdditionalInformation('pspReference', $response['pspReference']);
+            }
+
+            if (!empty($response['paymentData'])) {
+                $payment->setAdditionalInformation('adyenPaymentData', $response['paymentData']);
+            }
+
+            if (!empty($response['details'])) {
+                $payment->setAdditionalInformation('details', $response['details']);
+            }
+
             switch ($resultCode) {
-                case "IdentifyShopper":
-                    $payment->setAdditionalInformation('threeDSType', $resultCode);
-                    $payment->setAdditionalInformation(
-                        'threeDS2Token',
-                        $response['authentication']['threeds2.fingerprintToken']
-                    );
-                    $payment->setAdditionalInformation('adyenPaymentData', $response['paymentData']);
-                    break;
-                case "ChallengeShopper":
-                    $payment->setAdditionalInformation('threeDSType', $resultCode);
-                    $payment->setAdditionalInformation(
-                        'threeDS2Token',
-                        $response['authentication']['threeds2.challengeToken']
-                    );
-                    $payment->setAdditionalInformation('adyenPaymentData', $response['paymentData']);
-                    break;
                 case "Authorised":
                 case "Received":
+                    // TODO refactor since the full additionalData is stored in additionalInformation already
```

Comment:
@acampos1916 I think since we now use the component instead of using these additional data it can be removed from here as well

Reply:
Looks like it could be removed indeed. But maybe the setCcType should be kept

Reply:
Added it to the ticket PW-4496